x-pack/plugins/apm/server/routes/suggestions/get_suggestions_with_terms_aggregation.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import { rangeQuery, termQuery } from '../../../../observability/server';
+import { Setup } from '../../lib/helpers/setup_request';
+
+export async function getSuggestionsWithTermsAggregation({
+  fieldName,
+  fieldValue,
+  serviceName,
+  setup,
+  size,
+  start = moment().subtract(24, 'h').valueOf(),
+  end = moment().valueOf(),
+}: {
+  fieldName: string;
+  fieldValue: string;
+  serviceName: string;
+  setup: Setup;
+  size: number;
+  start?: number;
+  end?: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search(
+    'get_suggestions_with_terms_aggregation',
+    {
+      apm: {
+        events: [
+          ProcessorEvent.transaction,
```

Comment:
this is probably where we should use `searchAggregatedTransactions`. If we can avoid iterating over all the transaction events that'd be nice.

Reply:
updated: https://github.com/elastic/kibana/pull/126679/commits/85d56cbc9818d6bc4c8f962aae641bb5b35d4ed1